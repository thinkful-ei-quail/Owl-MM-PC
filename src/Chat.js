import React from 'react';
import './Chat.css';
import chatEvents from './chatEvents';
import participants from './participantsDummy';


function chatEvent(props) {
    // for each object in chatEvents participants are referenced by participantId, we want to display the participant Name

    //filter through participantsDummy and find where chatEvents.ParticipantId === participantsDummy.id
    let person = props.participants.find(person =>
        person.id === props.chatEvent.participantId);
    let name = <li>{person.name}</li>;
    let content = '';
    let date = new Date(props.chatEvent.timestamp);

    if(props.chatEvent.type === 'message') {
        content = <li>{date.toLocaleString()} {props.chatEvent.message}</li>
        }
    else {
        content = props.chatEvent.type;
    }
    

    return (
        <div>
            {name}
            {content}
        </div>)
}


export default chatEvent;