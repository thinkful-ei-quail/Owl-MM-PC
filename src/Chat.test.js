import React from 'react';
import ReactDOM from 'react-dom';
import Chat from './Chat';
import chatEvents from './chatEvents';
import participants from './participantsDummy';
import renderer from 'react-test-renderer';


it('should render chatlog', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Chat participants={participants} chatEvent={chatEvents[0]} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('snapshot of working? UI', () => {
    const tree = renderer 
        .create(<Chat participants={participants} chatEvent={chatEvents[0]}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});