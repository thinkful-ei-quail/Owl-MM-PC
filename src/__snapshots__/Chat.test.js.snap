// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of working? UI 1`] = `
<div>
  <li>
    Axe Kubicka
  </li>
  <li>
    1/30/2019, 5:48:04 AM
     
    Hello world
  </li>
</div>
`;
